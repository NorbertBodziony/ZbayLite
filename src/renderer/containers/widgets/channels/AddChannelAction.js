import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import AddChannelAction from '../../../components/widgets/channels/AddChannelAction'
import { actionCreators } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch => bindActionCreators({
  openCreateModal: actionCreators.openModal('createChannel')
}, dispatch)

export default connect(null, mapDispatchToProps)(AddChannelAction)
