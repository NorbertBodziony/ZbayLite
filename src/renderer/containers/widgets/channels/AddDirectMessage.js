import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import AddDirectMessage from '../../../components/widgets/channels/AddDirectMessage'
import { actionCreators } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch => bindActionCreators({
  openModal: actionCreators.openModal('sendMoney')
}, dispatch)

export default connect(null, mapDispatchToProps)(AddDirectMessage)
