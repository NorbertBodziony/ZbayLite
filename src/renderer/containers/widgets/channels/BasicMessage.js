import React from 'react'
import { connect } from 'react-redux'

import BasicMessage from '../../../components/widgets/channels/BasicMessage'
import channelSelectors from '../../../store/selectors/channel'
import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => {
  const signerPubKey = identitySelectors.signerPubKey(state)
  const channelOwner = channelSelectors.channelOwner(state)
  const shareableUri = channelSelectors.shareableUri(state)
  const channelModerators = channelSelectors.channelModerators(state)

  return {
    allowModeration: shareableUri
      ? channelOwner === signerPubKey || channelModerators.contains(signerPubKey)
      : false
  }
}

export default connect(mapStateToProps)(React.memo(BasicMessage))
