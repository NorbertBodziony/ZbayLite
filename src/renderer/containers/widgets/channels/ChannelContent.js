import { connect } from 'react-redux'

import ChannelContent from '../../../components/widgets/channels/ChannelContent'
import channelSelectors from '../../../store/selectors/channel'

export const mapStateToProps = state => ({
  loader: channelSelectors.loader(state),
  inputLocked: channelSelectors.inputLocked(state)
})

export default connect(mapStateToProps)(ChannelContent)
