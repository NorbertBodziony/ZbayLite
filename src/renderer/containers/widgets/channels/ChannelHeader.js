import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Immutable from 'immutable'

import ChannelHeader from '../../../components/widgets/channels/ChannelHeader'
import channelsHandlers from '../../../store/handlers/channels'
import notificationCenterHandlers from '../../../store/handlers/notificationCenter'

import channelSelectors from '../../../store/selectors/channel'
import contactsSelectors from '../../../store/selectors/contacts'
import identitySelectors from '../../../store/selectors/identity'
import notificationCenter from '../../../store/selectors/notificationCenter'

import { messageType, notificationFilterType } from '../../../../shared/static'

export const mapStateToProps = (state, props) => {
  const contact = contactsSelectors.contact(props.contactId)(state)
  return {
    channel: Immutable.fromJS({
      name: contact.get('username'),
      address: props.contactId
    }),
    userAddress: identitySelectors.address(state),
    members: channelSelectors.members(state),
    showAdSwitch: !!channelSelectors
      .messages()(state)
      .find(msg => msg.type === messageType.AD),
    mutedFlag:
      notificationCenter.channelFilterById(
        channelSelectors.data(state)
          ? channelSelectors.data(state).get('address')
          : 'none'
      )(state) === notificationFilterType.MUTE
  }
}
export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      updateShowInfoMsg: channelsHandlers.epics.updateShowInfoMsg,
      unmute: () =>
        notificationCenterHandlers.epics.setChannelsNotification(
          notificationFilterType.ALL_MESSAGES
        )
    },
    dispatch
  )
export default connect(mapStateToProps, mapDispatchToProps)(ChannelHeader)
