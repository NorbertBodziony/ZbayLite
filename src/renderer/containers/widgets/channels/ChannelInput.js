import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ChannelInputComponent from '../../../components/widgets/channels/ChannelInput'
import channelHandlers from '../../../store/handlers/channel'
import contactsHandlers from '../../../store/handlers/contacts'
import channelSelectors, { INPUT_STATE } from '../../../store/selectors/channel'
import usersSelectors from '../../../store/selectors/users'
import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = (state, { contactId }) => {
  const registered = contactId
    ? !!usersSelectors.registeredUser(identitySelectors.signerPubKey(state))(state)
    : true
  return {
    message: channelSelectors.message(state),
    inputState: registered ? channelSelectors.inputLocked(state) : INPUT_STATE.UNREGISTERED
  }
}

export const mapDispatchToProps = (dispatch, { contactId }) =>
  bindActionCreators(
    {
      onChange: channelHandlers.actions.setMessage,
      onKeyPress: contactId
        ? contactsHandlers.epics.sendDirectMessageOnEnter
        : channelHandlers.epics.sendOnEnter
    },
    dispatch
  )
export const ChannelInput = ({ onChange, onKeyPress, message, inputState }) => {
  const [infoClass, setInfoClass] = React.useState(null)
  return (
    <ChannelInputComponent
      infoClass={infoClass}
      setInfoClass={setInfoClass}
      onChange={onChange}
      onKeyPress={onKeyPress}
      message={message}
      inputState={inputState}
    />
  )
}

ChannelInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  onKeyPress: PropTypes.func.isRequired,
  inputState: PropTypes.number.isRequired,
  message: PropTypes.string
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ChannelInput)
