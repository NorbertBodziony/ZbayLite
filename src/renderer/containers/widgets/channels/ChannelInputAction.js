import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'

import { actionCreators } from '../../../store/handlers/modals'
import ChannelInputAction from '../../../components/widgets/channels/ChannelInputAction'
import directMessageSelectors from '../../../store/selectors/directMessageChannel'

export const mapDispatchToProps = (dispatch) => {
  return (
    bindActionCreators(
      {
        onPostOffer: () => console.warn('[ChannelInputAction] onPostOffer not implemented'),
        onSendMoney: (modalName, payload) => actionCreators.openModal(modalName, payload)()
      },
      dispatch
    )
  )
}

export const mapStateToProps = (state) => ({
  targetRecipientAddress: directMessageSelectors.targetRecipientAddress(state)
})

export default R.compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(ChannelInputAction)
