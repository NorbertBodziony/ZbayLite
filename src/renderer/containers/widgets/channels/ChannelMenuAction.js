import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'
import { withRouter } from 'react-router-dom'
import Immutable from 'immutable'
import ChannelMenuAction from '../../../components/widgets/channels/ChannelMenuAction'
import { actionCreators } from '../../../store/handlers/modals'
import importedChannelHandler from '../../../store/handlers/importedChannel'
import dmChannelSelectors from '../../../store/selectors/directMessageChannel'
import identitySelectors from '../../../store/selectors/identity'
import channelSelectors from '../../../store/selectors/channel'
import publicChannelsSelectors from '../../../store/selectors/publicChannels'

export const mapStateToProps = state => ({
  targetAddress: dmChannelSelectors.targetRecipientAddress(state),
  isOwner:
    channelSelectors.channelOwner(state) ===
    identitySelectors.signerPubKey(state),
  publicChannels: publicChannelsSelectors.publicChannels(state),
  channel: channelSelectors.data(state) || Immutable.Map({})
})

export const mapDispatchToProps = (dispatch, { history }) => {
  return bindActionCreators(
    {
      onInfo: actionCreators.openModal('channelInfo'),
      onMute: () => console.warn('[ChannelMenuAction] onMute not implemented'),
      onDelete: () => importedChannelHandler.epics.removeChannel(history),
      publishChannel: actionCreators.openModal('publishChannel'),
      onSettings: actionCreators.openModal('channelSettingsModal')
    },
    dispatch
  )
}

export default R.compose(
  withRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(ChannelMenuAction)
