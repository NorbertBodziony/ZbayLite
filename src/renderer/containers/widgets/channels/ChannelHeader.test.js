import Immutable from 'immutable'

import { mapStateToProps } from './ChannelHeader'

import create from '../../../store/create'
import { ChannelState } from '../../../store/handlers/channel'
import { ChannelsState } from '../../../store/handlers/channels'
import { createChannel } from '../../../testUtils'
import { NodeState } from '../../../store/handlers/node'

describe('ChannelHeader', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: Immutable.Map({
        channel: ChannelState({
          spentFilterValue: 38,
          id: 1
        }),
        node: NodeState({
          isTestnet: true
        }),
        channels: ChannelsState({
          data: Immutable.fromJS([
            createChannel(1),
            createChannel(2)
          ])
        })
      })
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })
})
