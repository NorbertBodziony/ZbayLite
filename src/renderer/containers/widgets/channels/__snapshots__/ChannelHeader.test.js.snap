// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelHeader will receive right props 1`] = `
Object {
  "channel": Immutable.Map {
    "id": 1,
    "name": "Channel 1",
    "private": true,
    "address": "zs1testaddress1",
    "unread": 1,
    "description": "Channel about 1",
    "keys": Immutable.Map {
      "ivk": "incoming-viewing-key-1",
    },
  },
  "members": Set {},
}
`;
